packages/block-editor/src/components/list-view/block-contents.js

```diff
@@ -61,23 +65,96 @@ const ListViewBlockContents = forwardRef(
 			? selectedClientIds
 			: [ clientId ];
 
+		const { blockName, blockAttributes } = useSelect(
+			( select ) => {
+				const blockObject =
+					select( blockEditorStore ).getBlock( clientId );
+				return {
+					blockName: blockObject?.name,
+					blockAttributes: blockObject?.attributes,
+				};
+			},
+			[ clientId ]
+		);
+
+		const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+		const metaDataSupport = getBlockSupport(
+			blockName,
+			'__experimentalMetadata',
+			false
+		);
+
+		const supportsBlockNaming = !! (
+			true === metaDataSupport || metaDataSupport?.name
+		);
+
+		const { isLocked } = useBlockLock( clientId );
+
+		const toggleLabelEditingMode = ( value ) => {
+			if ( ! supportsBlockNaming ) {
+				return;
+			}
+
+			setLabelEditingMode( value );
+		};
+
+		const blockTitle = useBlockDisplayTitle( {
+			clientId,
+			context: 'list-view',
+		} );
+
+		const blockInformation = useBlockDisplayInformation( clientId );
+
+		const className = classnames( 'block-editor-list-view-block-contents', {
+			'is-dropping-before': isBlockMoveTarget,
+			'has-block-naming-support': supportsBlockNaming,
+		} );
+
+		function inputSubmitHandler( updatedInputValue ) {
+			updateBlockAttributes( clientId, {
```

Comment:
Could we move this whole functionality around naming blocks to a hook `useBlockNaming` or something? It feels like we're adding a lot to `ListViewBlockContents` that is not owned by this component.